pdl_schema.md

```diff
@@ -0,0 +1,1105 @@
+and thus can be defined as top-level schemas. Named schemas can specify an
+optional namespace to avoid naming conflict between schemas with same name.
+The name prefixed with the namespace using the dot(.) separator becomes the
+fully qualified name(FQN) of the schema. Named schemas can be
+referenced from other schemas using the fully qualified name.
+
+Each top-level schema should be stored in its own file with a `.pdl` extension.
+Name of the file should match the schema name and the directory structure should
+match the namespace (similar to how Java classes are organized).
+
+The Pegasus code generator implements a resolver that is similar to Java class
+loaders. If there is a reference to a named schema, the code generator will try
+to look for a file in the code generator’s resolver path. The resolver path is
+similar to a Java classpath. The fully qualified name of the named schema will
+be translated to a relative file name. The relative file name is computed by
+replacing dots (“.”) in the fully qualified name by the directory path separator
+(typically “/”) and appending a `.pdl` extension. This relative file name looked
+up using each location in the resolver path until it finds a file that contains
+the named schema.
+
+
+## Record Type
+Records are the most common type of Pegasus schemas and usually the starting
+point when defining data models using pegasus. A record represents a Named
+entity with fields representing attributes of that entity. The fields can be
+primitive types, enums, unions, maps, arrays or other records.
+
+For example:
+```pdl
+namespace com.example.time
+
+record Date {
+  day: int
+  month: int
+  year: int
+}
+```
+The above example is defining a record called `Date`. This is defined in the
+namesapce `com.example.time`, giving it the fully qualified name
+`com.example.time.Date`. So this schema should be defined in the following file
+```
+<project-dir>/pegasus/com/example/time/Date.pdl
+```
+
+The `Date` record can then be referenced in other schemas:
+```pdl
+namespace com.example.models
+
+import com.example.time.Date
+
+record User {
+  firstName: string
+  birthDay: Date
+}
+```
+The above example is defining a record called `User`:
+* This record is using the namespace `com.example.models`. So the fully
+ qualified name is `com.example.models.User`.
+* This record is defining two fields, `firstName` and `birthDay`.
+ * `firstName` is a primitive field of type `string`.
+ * `birthDay` references the type `com.example.time.Date` from the first example
+* PDL supports Java like `import` feature that allows you to define references
+ to external types and then use them in the schema using their simple name.
+
+### Record Field attributes
+Pegasus supports additional features for defining the behavior of fields in a
+record.
+#### Record fields can be defined as optional.
+In pegasus, a field is required unless the field is explicitly declared as
```

Comment:
Pegasus should start with capital letter? If yes, please fix other places as well.

Reply:
This is a good question. I can never decide which way it is tbh

Reply:
I'm considering to update based on this rule:
 - Use "Pegasus" when the doc is talking about the Pegasus framework.
 - Use "pegasus" otherwise, eg, pegasus file, pegasus schema.

What do you think?

Reply:
Decided to use "Pegasus" everywhere